lsb_vsx/build.sh

```diff
@@ -0,0 +1,139 @@
+	if [ ! -d "${TET_ROOT}" ]; then
+	mkdir "${TET_ROOT}"
+	if [ ! -d "$TET_ROOT" ]; then
+		echo "Unable to create installation directory: ${TET_ROOT}"
+		exit 1
+	fi
+	fi
+
+	tet_package=${PREFIX_LSB_VSX}/tet_vsxgen_3.02.tgz
+	echo "Installing the TET and VSXgen test harness"
+	if [ ! -f "$tet_package" ]; then
+	echo "Could not find file \"$tet_package\""
+	echo Aborting
+	exit 1
+	fi
+
+	(cd "$TET_ROOT" && tar xfz "$tet_package")
+	if [ $? -ne 0 ]; then
+	echo "Unable to unpack TET/VSXgen"
+	exit 1
+	fi
+
+	# Install the test suites
+	cd "${PREFIX_LSB_VSX}" || exit
+	test_suites=$(echo lts_*.tgz | sed -e s/\ +//)
+	echo installing test_suites: "$test_suites"
+
+	current_dir=$PWD;
+	for test_suite in $test_suites; do
+	echo "Installing $test_suite"
+	(cd "$TET_ROOT"/test_sets && tar xfz "$current_dir"/"$test_suite")
+	if [ $? -ne 0 ]; then
+		echo "Unable to install test suite: $test_suite"
+	fi
+	done;
+
+	# Configure setup script and profiles
+	sed -e "s@^echo Unconfigured@TET_ROOT=$TET_ROOT@" "$TET_ROOT"/setup.sh.skeleton \
+	> "$TET_ROOT"/setup.sh && rm "$TET_ROOT"/setup.sh.skeleton
+	sed -e "s@^echo Unconfigured@TET_ROOT=$TET_ROOT@" "$TET_ROOT"/profile.skeleton \
+	> "$TET_ROOT"/profile && rm "$TET_ROOT"/profile.skeleton
+	sed -e "s@^echo Unconfigured@TET_ROOT=$TET_ROOT@" "$TET_ROOT"/test_sets/profile.skeleton  > "$TET_ROOT"/test_sets/profile && rm "$TET_ROOT"/test_sets/profile.skeleton
+fi
+
+#
+# # Apply patches
+#
+for patchfile in "$PREFIX_LSB_VSX_CONFIG"/patches/*.patch; do
+ 	if [ ! -f "$PREFIX_LSB_VSX_MARKERS/$(basename "$patchfile").applied" ]; then
+ 		echo "applying patch: $patchfile"
+ 		patch -d "$PREFIX_LSB_VSX_BUILD"/files -p1 < "$patchfile" 
+ 		touch "$PREFIX_LSB_VSX_MARKERS/$(basename "$patchfile").applied"
+ 	fi
+done
+
+#
+# # Build host executables needed to build tests
+#
+if [ ! -f "$PREFIX_LSB_VSX_MARKERS/host_step.stamp-files-prepared" ]; then
+	INCDIRS="/usr/include /usr/include/x86_64-linux-gnu"
+	CC=gcc
+	export INCDIRS
+	export CC
+	chmod 755 "$PREFIX_LSB_VSX_FILES"/setup.sh
+	(cd "$PREFIX_LSB_VSX_BUILD"/files && PLATFORM=HOST ./setup.sh)
+	#
+	# # Copy needed executables for later use
+	#
+	(cd "$PREFIX_LSB_VSX_FILES" && cp bin/* "$PREFIX_LSB_VSX_TMP"/tetbin_host)
+	(cd "$PREFIX_LSB_VSX_FILES" && cp test_sets/BIN/* "$PREFIX_LSB_VSX_TMP"/BIN_host)
+
+	touch "$PREFIX_LSB_VSX_MARKERS/host_step.stamp-files-prepared"
+
+	(cd "$PREFIX_LSB_VSX_FILES/src" && make clean)
+fi
+
+#
+# # Build all under phoenix
+#
+rm -f "$PREFIX_LSB_VSX_FILES"/test_sets/SRC/SYSINC/*
+rm "$PREFIX_LSB_VSX_FILES"/test_sets/SRC/SYSINC/sys/*
+rm "$PREFIX_LSB_VSX_FILES"/test_sets/SRC/SYSINC/rpc/*
+INCDIRS="$PREFIX_PROJECT/libphoenix/include"
+CC=$CC_BUF
+export LD
+export INCDIRS
+export CC
+(cd "$PREFIX_LSB_VSX_BUILD"/files && PLATFORM=PHOENIX-RTOS ./setup.sh)
+
+#
+# # Copy all to phoenix file system
+#
+mkdir -p "$PREFIX_ROOTFS/usr/test/lsb_vsx_posix"
+cp -a "$PREFIX_LSB_VSX_FILES" "$PREFIX_ROOTFS/usr/test/lsb_vsx_posix"
```

Comment:
aren't You copying too much here (I think there are also sources and intermediate object files being copied)?